src/Type/ParserNodeTypeToPHPStanType.php

```diff
@@ -58,6 +62,10 @@ public static function resolve($type, ?ClassReflection $classReflection): Type
 			}
 
 			return TypeCombinator::intersect(...$types);
+		} elseif ($type instanceof ConstFetch) {
+			return self::resolve(new Identifier($type->name->toString()), $classReflection);
```

Comment:
this path is taken, when a default is `null`. the `null` default value is passed as a `ConstFetch` for the name `'null'`.
therefore I turned it into a `Identifier` and let it be handled by existing logic.